src/plugins/shared_ux/public/components/page_template/no_data_page/no_data_card/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiCardProps } from '@elastic/eui';
+import { MouseEventHandler, ReactNode } from 'react';
+
+export type NoDataPageActions = Partial<EuiCardProps> & {
```

Comment:
Why separate the component type into a separate file?

Also since it's the direct props of NoDataCard, let's name it consistently:
```suggestion
export type NoDataCardProps = Partial<EuiCardProps> & {
```

Reply:
Because some of these types will be reused for `ElasticAgentCard` once I migrate it over, so that component can read from `types` file as well.